core/arch/aarchxx/mangle.c

```diff
@@ -603,6 +603,9 @@ insert_push_all_registers(dcontext_t *dcontext, clean_call_info_t *cci,
                                   DR_REG_LIST_LENGTH_ARM, DR_REG_LIST_ARM));
     }
     dstack_offs += 15 * XSP_SZ;
+
+    /* Make dstack_offs 8-byte algined, as we only accounted for 31 4-byte slots */
+    dstack_offs += XSP_SZ;
```

Comment:
Don't we need to actually adjust xsp also?  And then undo it in insert_pop_all_registers?